src/rust/engine/fs/src/directory.rs

```diff
@@ -811,6 +929,32 @@ impl From<&DigestTrie> for remexec::Tree {
   }
 }
 
+pub struct DigestTrieDiff {
+  our_unique_files: Vec<PathBuf>,
+  our_unique_dirs: Vec<PathBuf>,
+  their_unique_files: Vec<PathBuf>,
+  their_unique_dirs: Vec<PathBuf>,
+  changed_files: Vec<PathBuf>,
```

Comment:
Rather than exposing these fields with methods, it's probably fine to directly make the fields public by adding `pub`:
``suggestion
  pub changed_files: Vec<PathBuf>,
```

Reply:
Well the idea was that the functions each return an immutable vector reference, so outsiders can't mutate the items. But also meh.